### PR TITLE
created erlang sample app

### PR DESCRIPTION
**Pull request by mfelsche**
based upon the craterl Crate client and the cowboy webserver framework

### CODE REVIEW

**Comment by chaudum** [comment]
😳 will have a look asap! :)

**Comment by mfelsche** [comment]
* updated according to new spec
* tests run (against Crate 0.54.1)
* added documentation and license headers

File: erlang/src/guestbook_images_handler.erl [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  %% collection of images always exists
+  {true, Req, State};
+resource_exists(<<"POST">>, Req, #{craterl := CraterlClientRef, blob := Base64Blob}=State) ->
+  BinaryImage = base64:decode(Base64Blob),
+  %% find out if image already exists by actually uploading it
+  case guestbook_image_model:create_image(CraterlClientRef, BinaryImage) of
+    {ok, {created, HashDigest}} ->
+      {false, Req, State#{digest => HashDigest}};
+    {error, {already_exists, HashDigest}} ->
+      Req2 = guestbook_handlers:error_response(409, "Image with digest=\"" ++ binary_to_list(HashDigest) ++ "\ already exists", Req),
```
Comment (chaudum) [inline erlang/src/guestbook_images_handler.erl]: is here a `"` after `\` missing?

**Comment by chaudum** [comment]
otherwise LGTM (although I hardly understand half of the code 😏)

**Comment by chaudum** [comment]
integration tests are successful, so I'd say you can rebase, squash and merge :)

**Comment by mfelsche** [comment]
good opportunity to [learn you some erlang](learnyousomeerlang.com/) :)

**Comment by chaudum** [comment]
@mfelsche me looking at erlang code ...
confused haudi

**Pull request merged**